### PR TITLE
Make the healthcheck flags compatible with Docker CLI

### PR DESCRIPTION
**Pull request by csomh**
Docker CLI calls the healthcheck flags `--health-*`, instead of
`--healthcheck-*`. Introduce the former, in order to keep compatibility,
and alias the later, in order to avoid breaking current usage.

See discussion on #3455.

### CODE REVIEW

File: cmd/podman/common.go [hunk 1]
```diff
@@ -242,22 +242,42 @@ func getCreateFlags(c *cliconfig.PodmanCommand) {
 	createFlags.Bool(
 		"help", false, "",
 	)
+	createFlags.String(
+		"health-cmd", "",
+		"set a healthcheck command for the container ('none' disables the existing healthcheck)",
+	)
 	createFlags.String(
 		"healthcheck-command", "",
```
Comment (mheon) [inline cmd/podman/common.go]: I think Cobra allows aliases for flags - it'd be better to add these as aliases to the existing flags, not entirely new ones

Comment (csomh) [reply cmd/podman/common.go]: Can you point me to some docs, please? I've looked but could not find anything. This is why I've followed the same pattern that was used for the `--net`, `--network` pair.

(Cobra allows shorthands, as in `--publish, -p` with `StringP`, but that's not what is needed in this case.)

Comment (mheon) [reply cmd/podman/common.go]: I'm almost certain we used this somewhere... Let me poke around

Comment (mheon) [reply cmd/podman/common.go]: I might have been thinking of urfave/cli instead, which AFAIK allowed arbitrary numbers of flag aliases. Which, in cases like this, was a good bit more convenient...

File: cmd/podman/shared/create.go [hunk 2]
```diff
@@ -91,7 +91,11 @@ func CreateContainer(ctx context.Context, c *GenericCLIResults, runtime *libpod.
 
 		var healthCheckCommandInput string
 		// if the user disabled the healthcheck with "none", we skip adding it
-		healthCheckCommandInput = c.String("healthcheck-command")
+		healthCheckCommandInput = c.String("health-cmd")
```
Comment (mheon) [inline cmd/podman/shared/create.go]: I'd prefer we error here if both are set, instead of unconditionally overwriting

**Comment by rhatdan** [comment]
Inside of cobra they explain how to do this.

**Example #2**: You want to alias two flags. aka --old-flag-name == --new-flag-name

``` go
func aliasNormalizeFunc(f *pflag.FlagSet, name string) pflag.NormalizedName {
	switch name {
	case "old-flag-name":
		name = "new-flag-name"
		break
	}
	return pflag.NormalizedName(name)
}

myFlagSet.SetNormalizeFunc(aliasNormalizeFunc)
```

**Comment by csomh** [comment]
Thanks @rhatdan! That's what I was looking for.

I've updated the PR to use aliases instead. 

Varlink and GenericCLIResults were not changed, but I'm not entirely confident, that I understand those parts correctly, so let me know if those also need a change.

File: docs/podman-create.1.md [hunk 3]
```diff
@@ -268,26 +268,28 @@ The following example maps uids 0-2000 in the container to the uids 30000-31999
...
 
-**--healthcheck-command**=*command*
+**--health-cmd**, **--healthcheck-command**=*command*
 
 Set or alter a healthcheck command for a container. The command is a command to be executed inside your
 container that determines your container health.  The command is required for other healthcheck options
 to be applied.  A value of `none` disables existing healthchecks.
 
-**--healthcheck-interval**=*interval*
+Healthchecks currently only work when running as root.
```
Comment (mheon) [inline docs/podman-create.1.md]: Actually no longer the case - Giuseppe is fixing this with #3529

File: docs/podman-run.1.md [hunk 4]
```diff
@@ -275,26 +275,28 @@ The example maps gids 0-2000 in the container to the gids 30000-31999 on the hos
...
 
-**--healthcheck-command**=*command*
+**--health-cmd**, **--healthcheck-command**=*command*
 
 Set or alter a healthcheck command for a container. The command is a command to be executed inside your
 container that determines your container health.  The command is required for other healthcheck options
 to be applied.  A value of `none` disables existing healthchecks.
 
-**--healthcheck-interval**=*interval*
+Healthchecks currently only work when running as root.
```
Comment (mheon) [inline docs/podman-run.1.md]: Can also be removed through #3529

**Comment by mheon** [comment]
Slight touchups needed on the manpages, otherwise LGTM

File: docs/podman-create.1.md (hunk 3 above)
Comment (csomh) [reply docs/podman-create.1.md]: Removed these.

**Comment by mheon** [comment]
@csomh Mind rebasing? There are fixes for test failures on master

**Comment by csomh** [comment]
@mheon some are still failing. Would it make sense to rebase onto the latest green commit instead?

**Comment by mheon** [comment]
Checking, let me see if we're looking at flakes or real failures

**Comment by mheon** [comment]
```
[+2083s] panic: unable to find generic flag for varlink health-cmd
[+2083s] 
[+2083s] goroutine 1 [running]:
[+2083s] github.com/containers/libpod/cmd/podman/shared.GenericCLIResults.Find(...)
[+2083s] 	cmd/podman/shared/intermediate_varlink.go:433
[+2083s] github.com/containers/libpod/cmd/podman/shared.GenericCLIResults.MakeVarlink(0xc000390330, 0xc0003fd770, 0x2, 0x3, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, ...)
[+2083s] 	cmd/podman/shared/intermediate_varlink.go:107 +0x64b1
[+2083s] github.com/containers/libpod/pkg/adapter.(*LocalRuntime).Run(0xc0000ae048, 0x1a709a0, 0xc0000aa018, 0x2a07080, 0x7d, 0x0, 0x0, 0x0)
[+2083s] 	pkg/adapter/containers_remote.go:463 +0x77
[+2083s] main.runCmd(0x2a07080, 0x0, 0x0)
[+2083s] 	cmd/podman/run.go:60 +0x164
[+2083s] main.glob..func29(0x29a4180, 0xc0003fd770, 0x2, 0x3, 0x0, 0x0)
[+2083s] 	cmd/podman/run.go:23 +0x86
[+2083s] github.com/containers/libpod/vendor/github.com/spf13/cobra.(*Command).execute(0x29a4180, 0xc0000b4020, 0x3, 0x3, 0x29a4180, 0xc0000b4020)
[+2083s] 	vendor/github.com/spf13/cobra/command.go:826 +0x465
[+2083s] github.com/containers/libpod/vendor/github.com/spf13/cobra.(*Command).ExecuteC(0x29a9900, 0xc00009ccf0, 0x7fffb074ded9, 0x3e)
[+2083s] 	vendor/github.com/spf13/cobra/command.go:914 +0x2fc
[+2083s] github.com/containers/libpod/vendor/github.com/spf13/cobra.(*Command).Execute(...)
[+2083s] 	vendor/github.com/spf13/cobra/command.go:864
[+2083s] main.main()
[+2083s] 	cmd/podman/main.go:147 +0x88
```

**Comment by mheon** [comment]
I think we might have a real panic here

**Comment by csomh** [comment]
Of course we had... un-did the changes in `MakeVarlink()`.

**Comment by mheon** [comment]
Last failure looks like a flake, restarted

**Comment by rhatdan** [comment]
Command completions in contrib/bash/podman also need to be updated.

**Comment by csomh** [comment]
Do you mean `completions/bash/podman`?

`--health-start-period` was added to that one (the other `--health-*` flags were already there), and `--no-healthcheck` removed, as it's currently not an option in podman (`--health-cmd=none` might mean the same :confused:)

I did not update completion with the `--healhtcheck-*` variants, as I think this would make completing `--health` confusing (would bring up both variants, users might be confused which one to use), and might lead to users crafting incompatible commands as a result.

Let me know if you think otherwise.

**Comment by rhatdan** [comment]
Good on the completions. I agree on not exposing --healthcheck* at all.

Bottom line is remove it from being exposed anywhere but support it if a user used it within a script.

**Comment by csomh** [comment]
Removed `--healthcheck-*` from the docs and updated all the tests to use the documented flags.

**Comment by csomh** [comment]
@rhatdan @mheon could you please help me to re-run the failing tests? They don't seem to be related to the changes made. Prow does not seem to listen to my commands. Thanks!

**Comment by mheon** [comment]
Restarted them both - looked like known flakes

**Comment by rhatdan** [comment]
LGTM
@baude @mheon @giuseppe @vrothberg @ashley-cui @QiWang19 PTAL

**Comment by TomSweeneyRedHat** [comment]
LGTM and happy green test buttons.  TYVM @csomh !

**Comment by mheon** [comment]
Merging
/lgtm

**Pull request merged**